### DISCUSSION TITLE
Exposing set_hyper_builder within hyper 1.x HyperConnectorBuilder

### DISCUSSION THREAD

**Discussion by LeonHartley** [awslabs/aws-sdk-rust#1182] [General]
Is there a reason HyperConnectorBuilder for hyper 1.x doesn't publicly expose set_hyper_builder? Since this is experimental, is this something that will be exposed in the future? The hyper 0.14.x HyperConnectorBuilder exposes this publicly, so I am wondering why the 1.x version does not.

**Comment by aajtodd** [comment]
The client support for hyper was moved to `hyper-util` which is not 1.x. We are weighing what to expose still but don't want to end up in the same boat with hyper 0.x/1x we are now with `hyper-util`. 

What configuration would you like to have exposed? More than likely we could stabilize a minimal configuration surface without exposing hyper builder directly.

**Reply by LeonHartley** [reply]
We'd like the ability to override the TCP connector so we can use [Turmoil](https://github.com/tokio-rs/turmoil), that's the only use case I have in mind, at the moment we're using the hyper 0.14.x connector which allows us to override it, but gives us a bit of added complexity as we also reference hyper 1.x for everything else.
